managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerTest.java

```diff
@@ -1177,40 +1177,59 @@ public void testSetProperties() throws Exception {
         ledger.setProperties(properties);
         assertEquals(ledger.getProperties(), properties);
 
+        properties.put("key4", "value4");
+        ledger.setProperty("key4", "value4");
+        assertEquals(ledger.getProperties(), properties);
+
+        ledger.deleteProperty("key4");
+        properties.remove("key4");
+        assertEquals(ledger.getProperties(), properties);
+
         Map<String, String> newProperties = new HashMap<>();
-        newProperties.put("key4", "value4");
         newProperties.put("key5", "value5");
         newProperties.put("key6", "value6");
         ledger.setProperties(newProperties);
         assertEquals(ledger.getProperties(), newProperties);
     }
 
     @Test
-    public void testAsyncSetProperties() throws Exception {
-        final CountDownLatch latch = new CountDownLatch(1);
-        ManagedLedger ledger = factory.open("my_test_ledger");
-        Map<String, String> properties = new HashMap<>();
-        properties.put("key1", "value1");
-        properties.put("key2", "value2");
-        properties.put("key3", "value3");
-        ledger.setProperties(properties);
-        Map<String, String> newProperties = new HashMap<>();
-        newProperties.put("key4", "value4");
-        newProperties.put("key5", "value5");
-        newProperties.put("key6", "value6");
-        ledger.asyncSetProperties(newProperties, new AsyncCallbacks.SetPropertiesCallback() {
-            @Override
-            public void setPropertiesComplete(Map<String, String> properties, Object ctx) {
-                latch.countDown();
+    public void testConcurrentAsyncSetProperties() throws Exception {
+        final CountDownLatch latch = new CountDownLatch(1000);
+        ManagedLedger ledger = factory.open("my_test_ledger", new ManagedLedgerConfig().setMaxEntriesPerLedger(1));
+        Executor executor = Executors.newCachedThreadPool();
+        executor.execute(()->{
+            try {
+                for (int i = 0; i < 100; i++) {
+                    ledger.addEntry("data".getBytes(Encoding));
+                    Thread.sleep(300);
+                }
+            } catch (Exception e) {
+                fail(e.getMessage());
```

Comment:
Since this is not in the main thread, it will not fail the test itself.

We don't need to do it from different threads. It would be enough to do something like: 

```java
ledger.addEntry(e1);
ledger.setProperty("x", "xx");
ledger.addEntry(e2);
// ...
```

Reply:
Done.